quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerForJava.java

```diff
@@ -0,0 +1,351 @@
+
+	private JDTQuarkusManagerForJava() {
+		this.infos = new HashMap<>();
+	}
+
+	/**
+	 * Returns the codelens list according the given codelens parameters.
+	 * 
+	 * @param params  the codelens parameters
+	 * @param utils   the utilities class
+	 * @param monitor the monitor
+	 * @return the codelens list according the given codelens parameters.
+	 * @throws JavaModelException
+	 */
+	public List<? extends CodeLens> codeLens(QuarkusJavaCodeLensParams params, IJDTUtils utils,
+			IProgressMonitor monitor) throws JavaModelException {
+		String uri = params.getUri();
+		ITypeRoot typeRoot = resolveTypeRoot(uri, utils, monitor);
+		if (typeRoot == null) {
+			return Collections.emptyList();
+		}
+		IJavaElement[] elements = typeRoot.getChildren();
+		Collection<CodeLens> lenses = new LinkedHashSet<>(elements.length);
+		if (params.isUrlCodeLensEnabled()) {
+			int serverPort = getJDTQuarkusProjectInfo(typeRoot.getJavaProject()).getServerPort();
+			params.setLocalServerPort(serverPort);
+			collectURLCodeLenses(typeRoot, elements, null, lenses, params, utils, monitor);
+		}
+		if (monitor.isCanceled()) {
+			lenses.clear();
+		}
+		return new ArrayList<>(lenses);
+	}
+
+	/**
+	 * Given the uri returns a {@link ITypeRoot}. May return null if it can not
+	 * associate the uri with a Java file ot class file.
+	 *
+	 * @param uri
+	 * @param utils   JDT LS utilities
+	 * @param monitor the progress monitor
+	 * @return compilation unit
+	 */
+	private static ITypeRoot resolveTypeRoot(String uri, IJDTUtils utils, IProgressMonitor monitor) {
+		utils.waitForLifecycleJobs(monitor);
+		final ICompilationUnit unit = utils.resolveCompilationUnit(uri);
+		IClassFile classFile = null;
+		if (unit == null) {
+			classFile = utils.resolveClassFile(uri);
+			if (classFile == null) {
+				return null;
+			}
+		} else {
+			if (!unit.getResource().exists() || monitor.isCanceled()) {
+				return null;
+			}
+		}
+		return unit != null ? unit : classFile;
+	}
+
+	private JDTQuarkusProjectInfo getJDTQuarkusProjectInfo(IJavaProject project) throws JavaModelException {
+		JDTQuarkusProjectInfo info = infos.get(project);
+		if (info == null) {
+			info = new JDTQuarkusProjectInfo(project);
+			infos.put(project, info);
+		}
+		return info;
+	}
+
+	private void collectURLCodeLenses(ITypeRoot typeRoot, IJavaElement[] elements, String rootPath,
+			Collection<CodeLens> lenses, QuarkusJavaCodeLensParams params, IJDTUtils utils, IProgressMonitor monitor)
+			throws JavaModelException {
+		for (IJavaElement element : elements) {
+			if (monitor.isCanceled()) {
+				return;
+			}
+			if (element.getElementType() == IJavaElement.TYPE) {
+				IType type = (IType) element;
+				// Get value of JAX-RS @Path annotation from the class
+				String pathValue = getJaxRsPathValue(type);
+				if (pathValue != null) {
+					// Class is annotated with @Path
+					// Display code lens only if local server is available.
+					if (!params.isCheckServerAvailable()

```

Comment:
`params.isCheckServerAvailable()` seems to be always false because `params.setCheckServerAvailable()` is never called.  Therefore causing https://github.com/redhat-developer/vscode-quarkus/pull/146#issuecomment-558761502

Reply:
Indeed @xorye it should be fixed now.